lib/modules/manager/maven/update.ts

```diff
@@ -82,10 +82,18 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion as ReleaseType);
+    let releaseType = bumpVersion as ReleaseType;
+    if (semver.prerelease(currentValue)) {
+      releaseType = 'pre'.concat(bumpVersion) as ReleaseType;
+    }
+
+    let newPomVersion = semver.inc(currentValue, releaseType, 'SNAPSHOT');
     if (!newPomVersion) {
       throw new Error('semver inc failed');
     }
+    if (semver.prerelease(currentValue)) {
+      newPomVersion = newPomVersion.replace(/-SNAPSHOT.*/, '-SNAPSHOT');
```

Comment:
What's the reason for this replacement?

Reply:
the semver library will add -SNAPSHOT.0. But maven expects no numbers after -SNAPSHOT.

Reply:
Wouldn't you say that's a bug in the semver package then? The identifier we insert here is a perfectly valid `<pre-release>`, so I don't see why it must be converted into a `<pre-release identifier> "." <dot-separated pre-release identifiers>`.

Reply:
That's a good point. I was actually unsure about all the possible versioning mechanisms (since I mostly work with maven), so I thought it might be just a maven specialty. After your comment I looked up in the spec at semver.org. And if I understand the backus-naur form correct, then it should be absolutely fine to have either a pre-release-identifier or a pre-release identifiier with a dot-separated build.
So I also would see it as a bug in semver.

Now after studying semver's unit tests I have an idea WHY they implemented it that way. their "pre*" increment implementations also bumps a release-version to a pre-release. In this case it would not be defined whether it should create a prerelease version with or without build.

But I will nevertheless open an issue at node-semver.

Reply:
https://github.com/npm/node-semver/issues/441